sound/soc/sof/topology.c

```diff
@@ -146,11 +148,19 @@ static int sof_keyword_dapm_event(struct snd_soc_dapm_widget *w,
 	dev_dbg(sdev->dev, "received event %d for widget %s\n",
 		event, w->name);
 
+	/* get runtime PCM params using widget's stream name */
+	spcm = snd_sof_find_spcm_name(sdev, swidget->widget->sname);
+
 	/* process events */
 	switch (event) {
 	case SND_SOC_DAPM_PRE_PMU:
+		if (spcm->stream[stream].suspend_ignored) {
+			dev_dbg(sdev->dev, "PRE_PMU event ignored, KWD pipeline is already RUNNING\n");
+			return 0;
+		}
+
 		/* set pcm params */
-		ret = ipc_pcm_params(swidget, SOF_IPC_STREAM_CAPTURE);
+		ret = ipc_pcm_params(swidget, stream);
```

Comment:
Will it be more readable if we use SOF_IPC_STREAM_CAPTURE instead of stream?

Reply:
it'd make the test less readable

spcm->stream[stream].suspend_ignored -> spcm->stream[SNDRV_PCM_STREAM_CAPTURE].suspend_ignored

Reply:
yes, the macro SOF_IPC_STREAM_CAPTURE is too long and looks odd that why we are doing for capture only, that's why I change it here.